screens/Reader.js

```diff
@@ -409,8 +460,46 @@ class Reader extends React.Component {
     }, false);
       `;
   }
+  savePositionToProps(message){
+    let data=message.nativeEvent.data
+      let position=data.split('-')[1]
+
+      const startBaniList=JSON.parse(this.props.startBani)
+      if(position>0){
+      if(startBaniList.length==0){
+        this.currentBani.progress=position
+        startBaniList.push(this.currentBani)
+      }
+      else{
+        let isFound=false
+         startBaniList.forEach(element => {
+          if (element.id == this.currentBani.id) {
+             isFound=true
+          }
+        });
+        if(isFound){
+          startBaniList.forEach(element=>{
+            if(element.id==this.currentBani.id){
+
+              element.progress=position
+            }
+          })
```

Comment:
`Array.prototype.filter()` can be used here to reduce the code

Reply:
This function will create a new array with specified condition.  
However I just need to store the position if id matches. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter